src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -33,7 +33,7 @@ trait IterableLike[+A, +C[X] <: Iterable[X]]
   /** Create a collection of type `C[A]` from the elements of `coll`, which has
     *  the same element type as this collection. Overridden in StringOps and ArrayOps.
     */
-  protected[this] def fromIterableWithSameElemType(coll: Iterable[A]): C[A] = fromIterable(coll)
+  protected[this] def fromIterableWithSameElemType(coll: Iterable[A]): C[A]
```

Comment:
I removed this default implementation because in the case of `Map[K, V]`, `fromIterable` does not return a `Map[K, V]` but an `Iterable[(K, V)]`.